src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -479,18 +543,10 @@ class OrderedRVD(
     require(keep.isIncreasing && (keep.isEmpty || (keep.head >= 0 && keep.last < crdd.partitions.length)),
       "values not increasing or not in range [0, number of partitions)")
 
-    val newRangeBounds = Array.tabulate(keep.length) { i =>
-      if (i == 0)
-        partitioner.rangeBounds(keep(i))
-      else {
-        partitioner.rangeBounds(keep(i))
-          .copy(start = partitioner.rangeBounds(keep(i - 1)).end)
-      }
-    }
```

Comment:
any idea why this existed to begin with?

Reply:
This is fixing up the range bounds so that the right endpoint of one equals the left endpoint of the next. Until this PR, that was necessary for things to work correctly. Now gaps between partitions is perfectly fine.